### ISSUE TITLE
Plan gitlab_service_slack cannot unmarshal string into Go struct field SlackServiceProperties.properties.notify_only_broken_pipelines of type bool

### ISSUE THREAD

**Issue by meheza-samie** [gitlabhq/terraform-provider-gitlab#348]
Hi there,

Our automatic terraform pipeline broke since the last update of the gitlab API (2020-06-22).

### Terraform Version

- Terraform v0.12.26
- provider.gitlab: version = "~> 2.10"

### Affected Resource(s)
- gitlab_service_slack

### Terraform Configuration Files
```hcl

provider "gitlab" {
  token = var.GITLAB_ACCESS_TOKEN
}

resource "gitlab_service_slack" "this" {
  for_each = {
    for project in local.projects_settings : "${project.project_name}.${project.project_group_id}" => project
  }

  project                      = gitlab_project.this[each.key].id
  webhook                      = each.value.project_slack_webhook
  username                     = "GitLab"
  notify_only_broken_pipelines = true
  branches_to_be_notified      = "default_and_protected"
  push_events                  = false
  issues_events                = false
  confidential_issues_events   = false
  merge_requests_events        = true
  tag_push_events              = true
  note_events                  = false
  confidential_note_events     = false
  pipeline_events              = true
  wiki_page_events             = false
}

etc ...

```

### Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

### Expected Behavior
Terraform plan must not crash on gitlab_service_slack ressource.

### Actual Behavior

```
gitlab_branch_protection.master["**********"]: Refreshing state... [id=**********:master]
gitlab_branch_protection.master["**********"]: Refreshing state... [id=**********:master]
gitlab_branch_protection.master["**********"]: Refreshing state... [id=**********:master]
gitlab_branch_protection.master["**********"]: Refreshing state... [id=**********:master]

Error: json: cannot unmarshal string into Go struct field SlackServiceProperties.properties.notify_only_broken_pipelines of type bool

```
### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform plan`

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- https://github.com/terraform-providers/terraform-provider-gitlab/issues/255

**Comment by ymatsiuk** [comment]
@meheza-samie could you please add and example of `local.projects_settings`. I don't think it is possible to reproduce this issue only having resource `gitlab_service_slack`

**Comment by ymatsiuk** [comment]
https://www.terraform.io/docs/providers/gitlab/r/service_jira.html this is the list of supported options. `notify_only_broken_pipelines` is not in the list

**Comment by meheza-samie** [comment]
Hi @ymatsiuk thank your for your answer.
The issue is not on service_jira but in service_slack : gitlab_service_slack.

**Comment by ymatsiuk** [comment]
@meheza-samie sorry, I switched the context and lost the issue :smile: (wiped this message :blush: )

**Comment by meheza-samie** [comment]
@ymatsiuk please find below example of local.projects_settings : 

` projects_setting = [
    {  
       project_name             = "aws-teams-gitlab-test"
       project_description   = "repo for aws-teams-gitlab-test"
       project_group_id      = 1111111
       project_setting       =  {
        "lfs_enabled": false,
        "request_access_enabled": false,
        "only_allow_merge_if_all_discussions_are_resolved": true
        }
      project_runenv         = "tf"
      project_slack_webhook = "https://hooks.slack.com/services/*************************************"
      project_service_name  = "infra"
      }
]`

**Comment by meheza-samie** [comment]
Hi @ymatsiuk,

I have tried gitlab_service_slack in isolate terraform project and it seems to work.

```
provider "gitlab" {
  token = "*****************"
}

resource "gitlab_service_slack" "this" {
  project                      = 11111111
  webhook                      = "https://hooks.slack.com/services/*************************************"
  username                     = "GitLab"
  notify_only_broken_pipelines = true
  branches_to_be_notified      = "default_and_protected"
  push_events                  = false
  issues_events                = false
  confidential_issues_events   = false
  merge_requests_events        = true
  tag_push_events              = true
  note_events                  = false
  confidential_note_events     = false
  pipeline_events              = true
  wiki_page_events             = false
}
```

I am going to double check your project and let you know if I can find anything that can help you to reproduce.

**Comment by ymatsiuk** [comment]
that's the config I tried and it just works with gitlab `v13.0.6` terraform `v0.12.26` and provider.gitlab `v2.10.0`

```hcl
locals {
  projects_settings = {
    bar = {
      project_name          = "aws-teams-gitlab-bar"
      project_slack_webhook = "https://hooks.slack.com/"
    }
    foo = {
      project_name          = "aws-teams-gitlab-foo"
      project_slack_webhook = "https://hooks.slack.com/"
    }
  }
}

resource "gitlab_project" "slack" {
  for_each = {
    for project in local.projects_settings : project.project_name => project
  }
  name             = each.key
  description      = each.value.project_name
  visibility_level = "public"
}

resource "gitlab_service_slack" "slack" {
  for_each = {
    for project in local.projects_settings : project.project_name => project
  }
  project                      = gitlab_project.slack[each.key].id
  webhook                      = each.value.project_slack_webhook
  username                     = "GitLab"
  notify_only_broken_pipelines = true
  branches_to_be_notified      = "default_and_protected"
  push_events                  = false
  issues_events                = false
  confidential_issues_events   = false
  merge_requests_events        = true
  tag_push_events              = true
  note_events                  = false
  confidential_note_events     = false
  pipeline_events              = true
  wiki_page_events             = false
}
```

output: ```Plan: 4 to add, 0 to change, 0 to destroy.```

**Comment by meheza-samie** [comment]
Hi ,
I finally fix this issue with `terraform state rm 'gitlab_service_slack.this["**********"]'` on all gitlab_service_slack resources.

After that I compare the old tfstate with the new one and the only difference I can see is on the project tag of one project.

![image](https://user-images.githubusercontent.com/58169721/85713585-cc555100-b6e9-11ea-91a6-5807244e2ca2.png)

Best regards,

**Comment by schmidt-galen-heb** [comment]
Ran into this myself, and after some investigation I believe the root cause of the issue is the same as mentioned here: https://gitlab.com/gitlab-org/gitlab/-/issues/233941

In short, the Gitlab API is returning `"true"` instead of `true`, causing the type conversion to fail when the provider attempts to assign the `string` value to a `bool` field.

I've left a comment on the linked issue mentioning this affects the Slack Service API in addition to the Email Service API.

**Comment by kunallanjewar** [comment]
I have hit this issue recently as well and opened a PR here https://github.com/xanzy/go-gitlab/pull/919
Once it is merged, we can bump up the go-gitlab's version in this terraform provider.

**Comment by kunallanjewar** [comment]
Looks like my PR against the upstream is merged, so I'll wait until a release is created then bump latest version in this repo.

**Comment by pdecat** [comment]
Fixed since v3.0.0 via https://github.com/gitlabhq/terraform-provider-gitlab/commit/cf82021745dd9f6d448f144af51079ecf2787681#diff-37aff102a57d3d7b797f152915a6dc16R6

**Comment by armsnyder** [comment]
Thanks for confirming the fix @pdecat!

@meheza-samie I'm going to close this as fixed. Let us know if you're still having trouble using v3.0.0.
